src/k8s/pkg/k8sd/api/cluster_config.go

```diff
@@ -27,12 +27,19 @@ func (e *Endpoints) putClusterConfig(s *state.State, r *http.Request) response.R
 		return response.BadRequest(fmt.Errorf("failed to decode request: %w", err))
 	}
 
+	// Changing the datastore configuration is opt-in. We expect the caller to explicitly set the "external" type.
+	// The nil check is required to ensure we only fail if the DatastoreConfig is expected to change.
+	if req.DatastoreConfig.Type != nil && req.DatastoreConfig.GetType() != "external" {
+		return response.BadRequest(fmt.Errorf("failed to updated datastore config: type must be 'external' but is %s", req.DatastoreConfig.GetType()))
+	}
+
 	oldConfig, err := utils.GetClusterConfig(r.Context(), s)
 	if err != nil {
 		return response.InternalError(fmt.Errorf("failed to retrieve cluster configuration: %w", err))
 	}
 
 	requestedConfig := types.ClusterConfigFromUserFacing(req.Config)
+	requestedConfig.Datastore = types.DatastoreConfigFromUserFacing(req.DatastoreConfig)
```

Comment:
This always updates the `datastore` struct in the config, regardless if it's set in the request, right? But these will just be nil so we are OK